### PR TITLE
improve log interceptor

### PR DESCRIPTION
**Pull request by LucasMatheusDev**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I'm adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### Motivation 
When running the app on systems focused on application analysis, such as [BrowserStack](https://www.browserstack.com) , same in profile mode , the request logs can leak unwanted information, even if the developer has the option to remove the print, most of them are not aware of these "leaks" of requests.

This opens the way for an attacker who has an apk , aab or any other file that the browserStack
accept, be able to run the app on the website and view the logs

So I suggest the inclusion of this little code that by default only executes the prints in debug mode

### CODE REVIEW

**Review by AlexV525 — DISMISSED** [review]
This is a breaking change that previous users might also suffer from the change that logs are not printed by default, and I'm pretty sure that we can't make changes like this at the moment.
I'd suggest warning users when constructing the interceptor.

**Comment by LucasMatheusDev** [comment]
> This is a breaking change that previous users might also suffer from the change that logs are not printed by default, and I'm pretty sure that we can't make changes like this at the moment. I'd suggest warning users when constructing the interceptor.

Hello @AlexV525 , this change only prevents prints in Production mode, and prints in Debug mode remain by default.

Between avoiding and warning about a possible leak of sensitive data, I believe it's better to avoid.


I believe that no developer expects their app to show logs in Production mode, so this suggestion does not disrupt the workflow of those already using the package.

What do you think about it?

File: dio/lib/src/interceptors/log.dart [hunk 1]
```diff
@@ -132,3 +132,10 @@ class LogInterceptor extends Interceptor {
     msg.toString().split('\n').forEach(logPrint);
   }
 }
+
+void printOnlyDebug(Object? object) {
```
Comment (ueman) [inline dio/lib/src/interceptors/log.dart:136]:
```suggestion
void debugPrint(Object? object) {
```

**Comment by ueman** [comment]
I agree with @LucasMatheusDev. Though, we should enable people to use print even in production. We should also point out this change prominently in the changelog and readme.

Can you please add it to the changelog and the readme?

**Comment by AlexV525** [comment]
cc @ueman to review this again.

**Review by ueman — APPROVED** [review]
LGTM

**Pull request merged**